src/translator/translation_model.cpp

```diff
@@ -0,0 +1,177 @@
+#include "translation_model.h"
+
+#include "batch.h"
+#include "byte_array_util.h"
+#include "common/logging.h"
+#include "data/corpus.h"
+#include "data/text_input.h"
+#include "parser.h"
+#include "translator/beam_search.h"
+
+namespace marian {
+namespace bergamot {
+
+TranslationModel::TranslationModel(const Config &options, MemoryBundle &&memory /*=MemoryBundle{}*/,
+                                   size_t replicas /*=1*/)
```

Comment:
Some TODOs about how this is a horrible hack to be replaced when workspaces are extracted from graphs.